packages/common/src/TransactionUtils.ts

```diff
@@ -70,11 +88,24 @@ export const runTransaction = async ({
 
   // Compute the selected account nonce. If the account has a pending transaction then use the subsequent index after the
   // pending transactions to ensure this new transaction does not collide with any existing transactions in the mempool.
-  if (await accountHasPendingTransactions(web3, transactionConfig.from))
-    transactionConfig.nonce = await getPendingTransactionCount(web3, transactionConfig.from);
+  if (await accountHasPendingTransactions(augmentedWeb3, transactionConfig.from))
+    transactionConfig.nonce = await getPendingTransactionCount(augmentedWeb3, transactionConfig.from);
   // Else, there is no pending transaction and we use the current account transaction count as the nonce.
   // This method does not play nicely in tests. Leave the nonce null to auto fill.
-  else if (argv.network != "test") transactionConfig.nonce = await web3.eth.getTransactionCount(transactionConfig.from);
+  else if (argv.network != "test" && argv._.includes("test"))
+    transactionConfig.nonce = await augmentedWeb3.eth.getTransactionCount(transactionConfig.from);
+  // Store the transaction nonce in the augmentedWeb3 object so that it can be used in the future. This enables us to fire a
+  // bunch of transactions off without needing to wait for them to be included in the mempool by manually incrementing.
+  console.log("argv", argv);
+  if (argv.network != "test" && argv._.includes("test")) {
```

Comment:
What's the problem on test networks?

Reply:
we had this before. reason is really that sometimes we re-initalize web3 objects and sometimes this does not place well with the object storing it's nonce. without this a lot of tests fail. solution is to just let test run "normally" where they get the nonce from the provider.

Reply:
you are 100% right! fixed.